### ISSUE TITLE
Support for MonetDB

### ISSUE THREAD

**Issue by majidazimi** [jOOQ/jOOQ#4285]
MonetDB is yet another columnar RDBMS. It has JDBC version 4 driver but its SQL may not be as complete as MySQL or PostgreSQL.
https://www.monetdb.org

**Comment by lukaseder** [comment]
Monet... I suspect they want to compete with [Mondrian](http://community.pentaho.com/projects/mondrian/), another database named after a painter.

We can certainly keep this on the (very) long term roadmap, although, I doubt that we'll get a lot of value out of supporting this particular DB, it is not very popular: http://db-engines.com/en/ranking

**Comment by majidazimi** [comment]
we have an off line reporting application. We load data every month and run some heavy aggregation queries. The logic is not complex, just bunch of `sum()`, `avg()`, `max()`, `min()` and `GROUP BY` on a huge flat table. Due to sanctions against Iran, we can not buy commercial columnar databases such as HP vertica. Also our company can not afford an Oracle guy. But I should say the performance is blazing fast. A query containing a `GROUP BY` with multiple `avg()` over 2 billion records runs in 20 seconds on my development machine because of its parallel query processor.

**Comment by lukaseder** [comment]
Hmm, interesting. I would have just recommended Vertica, as we're going to support that in jOOQ 3.7: #1350, but I can see your point.

Other columnar databases, or at least databases that support column tables are:

- HANA (supported in jOOQ 3.6, perhaps SAP has different export regulations)
- RedShift (supported in jOOQ 3.7)
- SQL Server 2014
- PostgreSQL, with the relevant extensions, e.g. https://github.com/citusdata/cstore_fdw, even if the (biased) benchmark comparison doesn't look too good on the PostgreSQL side: https://www.monetdb.org/content/citusdb-postgresql-column-store-vs-monetdb-tpc-h-shootout)

As far as MonetDB is concerned, we'd have to look into adding support, but that might take a while as our priorities for jOOQ 3.7 are Vertica, Redshift, SQL Azure, and perhaps Teradata...

**Comment by majidazimi** [comment]
Actually It is not a major problem for us in the mean time. I have converted all queries in PostgreSQL side to JOOQ. I just wanted a unified query tool so I can convert MonetDB queries too and get rid of all string queries. By the way thank you for supporting.

**Comment by lukaseder** [comment]
Yes, it would be great if that would be possible. Well, perhaps it is already. There is the `JDBCDatabase` that you could try using for the source code generator. It uses JDBC's `DatabaseMetaData` for reverse engineering. And then, given that they claim to be very SQL standard compliant (https://www.monetdb.org/Documentation/Manuals/SQLreference/Features), you could try using either the PostgreSQL or the HSQLDB dialect. Those might work well for very simple queries...
